src/Voron/Impl/Compaction/StorageCompaction.cs

```diff
@@ -230,57 +230,97 @@ private static void CopyTrees(StorageEnvironment existingEnv, StorageEnvironment
             // Load table into structure 
             var inputTable = txr.OpenTable(schema, treeName);
 
-            // Create the new table, and replay inserts
-            using (var txw = compactedEnv.WriteTransaction())
-            {
-                schema.Create(txw, treeName);
-                var outputTable = txw.OpenTable(schema, treeName);
+            // The next three variables are used to know what our current
+            // progress is
+            var entriesLeft = inputTable.NumberOfEntries;
+
+            // It is very important that these slices be allocated in the
```

Comment:
When we iterate later on, we keep using the value of lastSlice throughout iterations that must stay alive beyond the WriteTransaction.

As an example, in line 309 we have: `lastSlice = schema.Key.GetSlice(txr.Allocator, entry);`. If this was `lastSlice = schema.Key.GetSlice(txw.Allocator, entry);` instead, the slice would not live beyond the WriteTransaction, and hence later accesses would be using a dangling pointer.

It may be better to move the comment to line 309 directly, as values from the iteration in line 261 are allocated directly in the read transaction due to how SeekForwardFrom works.